pkg/auth/authorizer/abac/abac.go

```diff
@@ -52,7 +53,46 @@ func (p policyLoadError) Error() string {
 type policyList []*api.Policy
 
 // TODO: Have policies be created via an API call and stored in REST storage.
-func NewFromFile(path string) (policyList, error) {
+func NewFromFile(path string) (*policyList, error) {
+
+	pl, err := newFromFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an empty watcher
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch the policy file
+	err = watcher.Add(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// On write operations on the policy file, reload policy file
+	go func() {
+		for {
+			select {
+			case ev := <-watcher.Events:
+				switch ev.Op {
+				case fsnotify.Write:
```

Comment:
Here are the operations fsnotify is able to monitor:
*CREATE: not relevant to us as this is watching for file/dir creation within a directory. We are watching one file only.
*WRITE: it is possible that multiple writes could happen consecutively, and we should only trigger a refresh on the last write
*REMOVE: stop watching the file and inode is gone
*RENAME: as inode of the file stays the same after a `mv`, it has no consequence for us. 
*CHMOD: not relevant to us

Some text editors like vim or emacs does funny things with files, e.g., When you do a `:wq` in vim, it renames the original file, writes a new file, and then deletes the original file. Emacs does similar things with its backup file prefixed with `~`. Besides periodic flushes from a text editor could cause a file to be written when it's half written, and a reload of the file could result in unintended behaviors.

The clean way this would work is if a copy of the policy file is obtained first, and then the copy is edited, and once done, it is finally copied over the original policy file. We need to make sure if this causes multiple write operations, only the last write triggers the file reload.